### PR TITLE
[BBC-9058] Storybook: Migrate Profile to MDX

### PR DESCRIPTION
**Pull request by Tuxosaurus**
Tested locally on Firefox Macos, also with canaray-release (only snapshot changes due to classes order)

Theme:
https://www.youtube.com/watch?v=FTxSXUzc96A

### CODE REVIEW

File: src/profile/profile.tsx [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 import { Rating } from '../rating'
 import { TextDisplayType } from '../text'
 import { TextBody } from '../typography/body'
+import { StyledProfile } from './Profile.style'
+
+// <Story> needs to have a JSX element wrapping
+// the component in the story otherwise the smart knobs
+// will override the props passed.
+// TODO remove after BBC-10918 is done (Storybook 6+)
+type DivProps = Readonly<{
```
Comment (gabrielagabriel) [inline src/profile/profile.tsx:15]: IMHO we should not use this hack just for story and we must keep Knobs

**Review by wakooka — CHANGES_REQUESTED** [review]

File: src/profile/profile.tsx [hunk 2]
```diff
@@ -1,12 +1,21 @@
 import React from 'react'
```
[the remark this answers is not in the document]
Comment (Tuxosaurus) [reply src/profile/profile.tsx:1]: Fixed

File: src/profile/Profile.story.mdx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  </Story>
+</Preview>
+
+## Specifications
+
+## Usage
+
+```jsx
+import { Profile } from '@blablacar/ui-library/build/profile'
+;<Profile title="Your profile title" />
```
[the remark this answers is not in the document]
Comment (Tuxosaurus) [reply src/profile/Profile.story.mdx]: I included all props, tbh I'm not sure that part is useful when stories have their own code/cpoy feature.

File: src/profile/Profile.unit.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ describe('Profile default', () => {
     expect(defaultProfile.find(Item).prop('aria-label')).toBeFalsy()
   })
 
-  it('Should pass a title prop to Item', () => {
+  it('Should pass change the leftTitleDisplay prop if isMedium', () => {
```
Comment (wakooka) [inline src/profile/Profile.unit.tsx:29]: typo: `pass change` ?

**Review by wakooka — APPROVED** [review]

File: src/profile/Profile.unit.tsx (hunk 4 above)
Comment (Tuxosaurus) [reply src/profile/Profile.unit.tsx]: Thanks

File: package.json [hunk 5]
```diff
@@ -42,7 +42,7 @@
   ],
   "license": "MIT",
   "devDependencies": {
-    "@babel/cli": "7.7.5",
+    "@babel/cli": "7.12.1",
```
Comment (gabrielagabriel) [inline package.json:45]: should not be part of this PR

File: src/profile/Profile.story.mdx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      info="Charism +10 (if no rating)"
+      picture="https://pbs.twimg.com/profile_images/749446875162505218/6r6-9wDn.jpg"
+    />
+  </Story>
+</Preview>
+
+## Medium
+
+<Preview>
+  <Story name="Medium">
```
Comment (gabrielagabriel) [inline src/profile/Profile.story.mdx:25]:
```suggestion
  <Story name="Size">
```

File: src/profile/Profile.story.mdx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      picture="https://pbs.twimg.com/profile_images/749446875162505218/6r6-9wDn.jpg"
+      isMedium
+    />
+  </Story>
+</Preview>
+
+## Ratings
+
+<Preview>
+  <Story name="Ratings">
```
Comment (gabrielagabriel) [inline src/profile/Profile.story.mdx:39]:
```suggestion
  <Story name="With ratings">
```

**Review by gabrielagabriel — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (Tuxosaurus) [reply package.json:45]: Should we rebase `mdx` to sync with `master`?

File: src/profile/Profile.story.mdx (hunk 6 above)
Comment (Tuxosaurus) [reply src/profile/Profile.story.mdx]: Changed

File: src/profile/Profile.story.mdx (hunk 7 above)
Comment (Tuxosaurus) [reply src/profile/Profile.story.mdx]: Changed

File: package.json (hunk 5 above)
Comment (Tuxosaurus) [reply package.json]: Alright I rebased this branch

**Review by mcellier — APPROVED** [review]
